File: app/src/main/java/com/example/navigationdrawerapp/DrawerModel.java
package com.example.navigationdrawerapp;

import androidx.annotation.NonNull;

public class DrawerModel {
    private final String text;
    private final Boolean isHeader;
    private Boolean isExpand;
    public DrawerModel(String text, Boolean isHeader, Boolean isExpand) {
        this.text = text;
        this.isHeader = isHeader;
        this.isExpand = isExpand;
    }

    public String getText() {
        return text;
    }

    public Boolean getHeader() {
        return isHeader;
    }

    public Boolean getExpand() {
        return isExpand;
    }

    public void setExpand(Boolean expand) {
        isExpand = expand;
    }

    @NonNull
    @Override
    public String toString() {
        return "DrawerModel text " + text + " isHeader " + isHeader + " isExpand " + isExpand;
    }
}

File: app/src/main/java/com/example/navigationdrawerapp/DrawerModel.kt
package com.example.navigationdrawerapp

import androidx.compose.runtime.Immutable

@Immutable
public data class DrawerModel (
    val text : String,
    val isHeader : Boolean,
    var isExpand : Boolean,
    val icon : Int,
) {

    companion object {
        private val TAG = DrawerModel::class.java.getSimpleName()
    }

    override fun toString() : String {
        return "$TAG text $text isHeader $isHeader isExpand $isExpand" ?: super.toString()
    }
}